datasafe-encryption/datasafe-encryption-impl/src/main/java/de/adorsys/datasafe/encrypiton/impl/keystore/KeyStoreServiceImpl.java

```diff
@@ -62,39 +70,39 @@ public KeyStore createKeyStore(KeyStoreAuth keyStoreAuth,
         // TODO, hier also statt der StoreID nun das
         String serverKeyPairAliasPrefix = UUID.randomUUID().toString();
         log.debug("keystoreid = {}", serverKeyPairAliasPrefix);
-        KeyStoreGenerator keyStoreGenerator = KeyStoreGenerator.builder()
-                .keyCreationConfig(keyConfig)
-                .keyStoreConfig(config)
-                .serverKeyPairAliasPrefix(serverKeyPairAliasPrefix)
-                .readKeyPassword(keyStoreAuth.getReadKeyPassword())
-                .secretKeys(secretKeys)
+
+        EncryptingKeyCreationCfg encConf = keyConfig.getEncrypting();
+        Supplier<char[]> passSupplier = () -> keyStoreAuth.getReadKeyPassword().getValue();
+        KeySetTemplate template = KeySetTemplate.builder()
+                .generatedEncryptionKeys(Encrypting.with()
+                        .algo(encConf.getAlgo())
+                        .sigAlgo(encConf.getSigAlgo())
+                        .keySize(encConf.getSize())
+                        .prefix("ENC").password(passSupplier)
+                        .build()
+                        .repeat(keyConfig.getEncKeyNumber()))
+                .generatedSecretKeys(secretKeys.keySet().stream()
+                        .map(it -> Secret.with().prefix(it.getValue()).build())
+                        .collect(Collectors.toList()))
                 .build();
+        KeySet keySet = juggler.generateKeys().fromTemplate(template);
 
-        KeyStore userKeyStore = keyStoreGenerator.generate();
+        KeyStore ks = juggler.toKeystore().withConfig(config).generate(keySet, passSupplier);
```

Comment:
```.withConfig(config)```
Is unnecessary - we already have default in Juggler
See 
https://github.com/adorsys/datasafe/pull/181/files#diff-88b877cc34976bb2cfb6d55188a7768eR39

Reply:
removed